distributed/tests/test_s3.py

```diff
@@ -393,3 +394,91 @@ def test_read_csv_sync(loop):
             assert list(df.columns) == ['name', 'amount', 'id']
             f = e.compute(df.amount.sum())
             assert f.result() == (100 + 200 + 300 + 400 + 500 + 600)
+
+def test_avro_body(s3):
+    fastavro = pytest.importorskip('fastavro')
+    fn = 'distributed-test/test/data/avro/2.avro'
+    byte = s3.cat(fn)
+    av = fastavro.reader(s3.open(fn))
+    header = av._header
+    records = list(av)
+    
+    schema = header['meta']['avro.schema'].decode()
+    schema = json.loads(schema)
+    assert ['amount', 'name'] == sorted(f['name'] for f in schema['fields'])
+    sync = header['sync']
+    assert len(sync) == 16
+    bit = byte[byte.index(sync)+len(sync):]
+    assert avro_body(bit, header) == records
+    bit = byte[byte.index(sync)+len(sync):-len(sync)]
+    assert avro_body(bit, header) == records
+    assert avro_body(b"", header) == []
```

Comment:
Ideally we would be able to test this function without relying on S3.  Perhaps now would be a good time to think about refactoring out formats?  We're not particularly time constrained on the avro work (no one is actively asking for it) so the choice between copy-paste and refactor might change here (we may want to refactor.)